iocore/net/TLSSNISupport.cc

```diff
@@ -130,8 +130,56 @@ TLSSNISupport::on_servername(SSL *ssl, int *al, void *arg)
   this->_fire_ssl_servername_event();
 
   const char *name = SSL_get_servername(ssl, TLSEXT_NAMETYPE_host_name);
+
   if (name) {
     this->_set_sni_server_name(name);
+
+#if TS_HAS_TLS_EARLY_DATA
```

Comment:
I don't want to have code for early data here. One or two lines maybe acceptable if it's really needed. And DRY.